textworld/generator/chaining.py

```diff
@@ -406,6 +422,9 @@ def apply(self, node: _Node, action: Action) -> Optional[State]:
         new_state.apply(action)
 
         # Some debug checks
+        # XXX
+        if not self.check_state(new_state):
```

Comment:
@tavianator forget what I said about removing the first `check_state` (line 419). Removing it would cause duplicate in the chains we generate. Let's just remove the assert then.